app/models/result.rb

```diff
@@ -85,8 +85,8 @@ def get_subtotal(assignment: nil)
       else
         user_visibility = :ta_visible
       end
-      criteria = assignment.criteria.where(user_visibility => true).map(&:id)
-      marks_array = (marks.to_a.select { |m| criteria.member? m.criterion_id }).map(&:mark)
+      criterion_ids = assignment.criteria.where(user_visibility => true).ids
+      marks_array = (marks.to_a.select { |m| criterion_ids.member? m.criterion_id }).map(&:mark)
```

Comment:
So looking down a bit, it seems that this is similar to the "max mark" calculation we had before. You should be able to reduce this into a single query to return the subtotal now, something like `marks.where(criterion_id: criterion_ids).sum`